glance_adoption.md

```diff
@@ -0,0 +1,240 @@
+'
+```
+
+However, if a Ceph backend is used, the `customServiceConfig` parameter should
+be used to inject the right configuration to the `GlanceAPI` instance.
+
+Make sure the Ceph related secret exists in the `openstack` namespace:
+
+```
+$ oc get secrets | grep ceph
+ceph-conf-files
+```
+
+If it doesn't exist, create a `Secret` which contains the `Cephx` key and Ceph
+configuration file so that the Glance pod created by the operator can mount
+those files in `/etc/ceph`.
+
+```
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: ceph-client-conf
+  namespace: openstack
+stringData:
+  ceph.client.openstack.keyring: |
+    [client.openstack]
+        key = <secret key>
+        caps mgr = "allow *"
+        caps mon = "profile rbd"
+        caps osd = "profile rbd pool=images"
+  ceph.conf: |
+    [global]
+    fsid = 7a1719e8-9c59-49e2-ae2b-d7eb08c695d4
+    mon_host = 10.1.1.2,10.1.1.3,10.1.1.4
+```
+
+This secret will be used in the `extraVolumes` parameters to propagate the files
+to the `GlanceAPI` pods (both internal and external).
+Edit the `osp-secret` and change the `GlancePassword` to make sure the service
+will be able to interact with keystone, which points to the credentials of the
+source Cloud.
+
+```
+$ cat tripleo-standalone-passwords.yaml  | awk '/Glance/ { print $2 }'
+Ok1gAcO2IsHBsUUW6bAyC5gcA
+```
+
+The `base64` value of this secret should be used to patch the `osp-secret`.
+
+Patch OpenStackControlPlane to deploy Glance:
+
+```
+oc patch openstackcontrolplane openstack --type=merge --patch '
+spec:
+  glance:
+    enabled: true
+    template:
+      databaseInstance: openstack
+      containerImage: quay.io/tripleozedcentos9/openstack-glance-api:current-tripleo
+      customServiceConfig: |
+        [DEFAULT]
+        enabled_backends=default_backend:rbd
+        [glance_store]
+        default_backend=default_backend
+        [default_backend]
+        rbd_store_ceph_conf=/etc/ceph/ceph.conf
+        rbd_store_user=openstack
+        rbd_store_pool=images
+        store_description=Ceph glance store backend.
+      storageClass: "local-storage"
+      storageRequest: 10G
+      glanceAPIInternal:
+        containerImage: quay.io/tripleozedcentos9/openstack-glance-api:current-tripleo
+      glanceAPIExternal:
+        containerImage: quay.io/tripleozedcentos9/openstack-glance-api:current-tripleo
+  extraMounts:
+    - extraVol:
+      - propagation:
+        - Glance
+        extraVolType: Ceph
+        volumes:
+        - name: ceph
+          projected:
+            sources:
+            - secret:
+                name: ceph-conf-files
+        mounts:
+        - name: ceph
+          mountPath: "/etc/ceph"
+          readOnly: true
+'
+```
+
+## Test the glance service from the OpenStack cli
+
+Inspect the resulting glance pods:
+
+```
+sh-5.1# cat /etc/glance/glance.conf.d/01-custom.conf
+
+[DEFAULT]
+enabled_backends=default_backend:rbd,ceph1:rbd
```

Comment:
This should be defined only once

Reply:
+1, bad paste, I'll fix it in a follow up!